source/rules/roles.txt

```diff
@@ -185,12 +183,28 @@ data from the perspective of your app's users and what it might mean to them.
 How App Services Assigns Roles
 ------------------------------
 
-App Services dynamically assigns roles for every request. The user
-is assigned a separate role, or no role, for each document that matches
-the incoming query.
+App Services assigns roles differently depending on whether you're using
+:ref:`Device Sync (Flexible Mode) <sync>` or not.
+
+When using Device Sync, App Services assigns roles at the start of a "sync
+session"; otherwise, App Services assigns roles on a per-document basis. A sync
+session starts when opening a sync connection and ends when closing the sync
+connection.
 
-First, your App evaluates and applies :ref:`filters <filters>` and then
-runs the query.
+You can define a set of roles for a collection and default roles for when the
+collection roles don't apply. To assign a role, App Services evaluates each
+role's "apply when" expression. The first role whose "apply when" expression
+evaluates to true becomes the assigned role.
+
+Role Assignment: Device Sync Disabled
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

Comment:
[scannability] This heading and its counterpart are very similar, with the only difference being a single word at the end of the heading. This makes scanning harder.

What do you think about these suggestions?

Putting the differentiator at the beginning:
- Enabled Device Sync and Role Assignment
- Disabled Device Sync and Role Assignment

Differentiator first and shortened (though abstracted):
- With Device Sync
- Without Device Sync

Shortening the headings so you can scan to the end faster:
- Device Sync Enabled
- Device Sync Disabled